orc8r/gateway/python/scripts/register.py

```diff
@@ -0,0 +1,141 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import argparse
+import os
+import sys
+import stat
+import textwrap
+from typing import List
+
+import snowflake
+from magma.common.cert_utils import load_public_key_to_base64der
+from orc8r.protos.bootstrapper_pb2 import (
+    RegisterRequest,
+    ChallengeKey,
+    RegisterResponse,
+)
+from orc8r.protos.identity_pb2 import AccessGatewayID
+from orc8r.protos.bootstrapper_pb2_grpc import RegistrationStub
+from magma.common.service_registry import ServiceRegistry
+
+
+def register_handler(client: RegistrationStub, args: List[str]) -> RegisterResponse:
+    """
+    Registers a device and retrieves its control proxy
+    
+    Returns:
+        RegisterResponse: response from gRPC call, either error or the control_proxy
+    """
+    req = RegisterRequest(
+        token=args.token,
+        hwid=AccessGatewayID(
+            id=snowflake.snowflake()
+        ),
+        challenge_key=ChallengeKey(
+            key=load_public_key_to_base64der("/var/opt/magma/certs/gw_challenge.key")
+        ),
+    )
+
+    res = client.Register(req)
+    if res.HasField("error"):
+        raise Exception(res.error)
+
+    msg = textwrap.dedent(
+        """
+        > Successfully registered gateway
+        Hardware ID 
+        -----------
+        {}
+        Challenge Key
+        -----------
+        {}
+        Control Proxy
+        -----------
+        {}
+        """
+    )
+    print(msg.format(req.hwid, req.challenge_key, res.control_proxy))
+
+    return res
+
+
+def write_control_proxy(control_proxy: str):
+    """
+    Provisions the gateway with the control proxy
+
+    Returns:
+        None
+    """
+    filename = "/var/opt/magma/configs/control_proxy.yml"
+    os.system("sudo mkdir %s" % os.path.dirname(filename))
+    os.system("sudo touch %s" % filename)
+    os.system("sudo chmod 777 %s" % filename)
+
+    with open(filename, 'w') as control_proxy_f:
+        control_proxy_f.write(control_proxy)
+    control_proxy_f.close()
+
+
+def main():
+    """Register a gateway"""
+    parser = argparse.ArgumentParser(description="Register a gateway.")
+    parser.add_argument(
+        "domain",
+        metavar="DOMAIN_NAME",
+        type=str,
+        help="orc8r's domain name",
+    )
+    parser.add_argument(
+        "token",
+        metavar="REGISTRATION_TOKEN",
+        type=str,
+        help="registration token after API call",
+    )
+    parser.add_argument(
+        "--ca_file",
+        type=str,
+        help="orc8r's root CA file"
+    )
+    parser.add_argument(
+        "--no_control_proxy",
+        action="store_true",
+        help="disables writing the control proxy file"
+    )
+    args = parser.parse_args()
+
+    chan = ServiceRegistry.get_bare_bootstrap_rpc_channel(
+        args.domain,
+        '7444' if args.domain == 'magma.test' else '8444',
```

Comment:
not sure if there is a better way to tell whether we are in test env or not??

Reply:
I'd say let's make it a CLI arg, default to 8444. Then in test env you have to pass e.g. `--cloud-port 7444`